Core/src/Surfaces/LineSurface.cpp

```diff
@@ -80,26 +79,21 @@ Acts::Result<Acts::Vector2> Acts::LineSurface::globalToLocal(
     const GeometryContext& gctx, const Vector3& position,
     const Vector3& direction, double tolerance) const {
   using VectorHelpers::perp;
-  const auto& sTransform = transform(gctx);
-  const auto& tMatrix = sTransform.matrix();
-  Vector3 lineDirection(tMatrix(0, 2), tMatrix(1, 2), tMatrix(2, 2));
+
+  Transform3 totalTransform =
+      referenceFrame(gctx, position, direction).inverse() *
+      transform(gctx).inverse();
```

Comment:
Also here I am not sure if I understand the difference between the old and the new code